lib/db_utils.pm

```diff
@@ -75,7 +75,7 @@ sub influxdb_push_data {
     $out =~ s/$token/<redacted>/;
     record_info('curl POST', $out);
 
-    my $output = script_output($cmd, quiet => $args{quiet});
+    my $output = script_output($cmd, quiet => $args{quiet}), $args{proceed_on_failure});
```

Comment:
This will still not work, because `$return_code` will be undefined in the next line and then in line 80 the `unless` will be executed.

In short - this alone is insufficient, the test will still fail on timeouts.

To solve this you could use `script_run` to run `$cmd` and pipe the output to a temporary file. Then you can use `script_output` to extract the contents of that file. Like this you can check the return code and get the output.